src/Evolution/Executables/GrMhd/ValenciaDivClean/EvolveValenciaDivClean.hpp

```diff
@@ -112,8 +116,12 @@ struct EvolutionMetavars {
       grmhd::ValenciaDivClean::PrimitiveRecoverySchemes::NewmanHamlin,
       grmhd::ValenciaDivClean::PrimitiveRecoverySchemes::PalenzuelaEtAl>;
 
-  using observed_reduction_data_tags = observers::collect_reduction_data_tags<
-      tmpl::list<dg::Events::Observe<system>>>;
+  using observed_reduction_data_tags =
+      observers::collect_reduction_data_tags<tmpl::list<dg::Events::Observe<
+          3,
+          tmpl::append<system::variables_tag::tags_list,
+                       system::primitive_variables_tag::tags_list>,
+          system::primitive_variables_tag::tags_list>>>;
```

Comment:
you could make the observe event a type alias to remove the code duplication here. to avoid confusion it would probably be good to make that private, or add a comment that it's not part of the public interface

Reply:
Actually, just make it `observers::collect_reduction_data_tags<events>`.

Reply:
It’s OK for the metafunction if not all the events are observers, it works similar to the one that collects `const_global_cache_tags` from an action list. I see now that this instantiates differen types, but it would be good to remove the code duplication nontheless. If that’s not possible, so be it

Reply:
Ah, OK.  Since it is fine to pass extra stuff to the metafunction I think I found a cleaner way to do this.